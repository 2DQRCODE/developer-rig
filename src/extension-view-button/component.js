import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './component.sass'

export class ExtensionViewButton extends Component {
  render() {
    return (
      <div className="circle-button" onClick={this.props.onClick}>
        <p>+</p>
      </div>
    )
  }
}

ExtensionViewButton.propTypes = {
  onClick: PropTypes.func.isRequired
}
