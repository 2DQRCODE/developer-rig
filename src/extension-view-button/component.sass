.circle-button
  border-width: thick
  border-radius: 50%
  width: 50px
  height: 50px
  margin: 0 auto
  margin-top: 10px
  border: 2px solid #DAD8DE
  position: relative
  font-weight: bold
  display: table
  cursor: pointer

.circle-button p
  text-align: center
  vertical-align: middle
  display: table-cell

