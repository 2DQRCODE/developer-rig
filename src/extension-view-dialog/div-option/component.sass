@import '../component.sass'
$dimension: 70px

.extension-container__block
  @extend .container__block

  .block__extension-type-label
    visibility: hidden
    position: absolute
    cursor: pointer

  .extension-type-label__extension-type-text--color-black
    @extend .extension-type-text

  .extension-type-label__extension-type-text--color-purple
    @extend .extension-type-text
    color: #7D55C7

.extension-type-box
  margin-top: 10px
  margin-right: 12px
  height: $dimension
  width: $dimension
  border-radius: 5px
  border: 2px solid black
  display: table-cell
  vertical-align: middle
  text-align: center

.extension-type-text
    margin: 3px
    font-size: 10px
    text-align: center
    width: $dimension
