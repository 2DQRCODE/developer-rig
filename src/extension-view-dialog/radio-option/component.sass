.option-label
  clear: left
  font-size: 12px
  margin-right: 10px
