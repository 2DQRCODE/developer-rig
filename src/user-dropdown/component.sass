.user-dropdown
  display: flex
  flex-grow: 1
  justify-content: flex-end
  align-items: center
  padding-right: 1rem
  font-size: 13px

.user-dropdown__image
  width: 20px
  height: 20px

.user-dropdown__username
  margin-left: 5px
