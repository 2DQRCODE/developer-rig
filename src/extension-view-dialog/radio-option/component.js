import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';

export class RadioOption extends Component {
  render() {
    return (
      <div>
        <input type="radio" name={this.props.name} value={this.props.value}  onChange={this.props.onChange} defaultChecked={this.props.checked}/>
        <label className="option-label"> {this.props.value} </label>
      </div>
    )
  }
}

RadioOption.propTypes = {
  name: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  checked: PropTypes.bool
}
