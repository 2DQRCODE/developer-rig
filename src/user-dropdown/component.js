import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './component.sass';

export class UserDropdown extends Component {
  render() {
    const { login, profile_image_url } = this.props.login;
    return (
      <div className='user-dropdown'>
        <img alt='profile' className='user-dropdown__image' src={profile_image_url} />
        <div className='user-dropdown__username'>{login}</div>
      </div>
    );
  }
}

UserDropdown.propTypes = {
  login: PropTypes.object,
};
