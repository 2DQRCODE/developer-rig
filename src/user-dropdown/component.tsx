import * as React from 'react';
import { UserSession } from '../core/models/user-session';
import './component.sass';

export interface PublicProps {
  session: UserSession;
}
type Props = PublicProps;
export class UserDropdown extends React.Component<Props>{
  public render() {
    if (!this.props.session) {
      return null;
    }
    const { login, profileImageUrl } = this.props.session;
    return (
      <div className='user-dropdown'>
        <img alt='profile' className='user-dropdown__image' src={profileImageUrl} />
        <div className='user-dropdown__username'>{login}</div>
      </div>
    );
  }
}
