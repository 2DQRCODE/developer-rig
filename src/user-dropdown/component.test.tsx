import { setupShallowTestWithStore, setupShallowTest } from '../tests/enzyme-util/shallow';
import { UserDropdown } from './component';
import { EXTENSION_VIEWS, BROADCASTER_CONFIG, LIVE_CONFIG, CONFIGURATIONS  } from '../constants/nav-items';

describe('<UserDropdown />', () => {
  const defaultGenerator = () => ({
    session: { login: 'test', profileImageUrl: 'test.png', authToken: 'test' },
  });
  const setupRenderer = setupShallowTest(UserDropdown, defaultGenerator);

  it('renders correctly', () => {
    const { wrapper } = setupRenderer();
    expect(wrapper).toMatchSnapshot();
  });

  it('does not render if session null', () => {
    const { wrapper } = setupRenderer({
      session: null,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
