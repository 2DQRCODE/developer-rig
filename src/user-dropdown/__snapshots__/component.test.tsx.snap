// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDropdown /> does not render if session null 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserDropdown
    session={null}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<UserDropdown /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserDropdown
    session={
        Object {
            "authToken": "test",
            "login": "test",
            "profileImageUrl": "test.png",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="profile"
          className="user-dropdown__image"
          src="test.png"
/>,
        <div
          className="user-dropdown__username"
>
          test
</div>,
      ],
      "className": "user-dropdown",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "profile",
          "className": "user-dropdown__image",
          "src": "test.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "className": "user-dropdown__username",
        },
        "ref": null,
        "rendered": "test",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="profile"
            className="user-dropdown__image"
            src="test.png"
/>,
          <div
            className="user-dropdown__username"
>
            test
</div>,
        ],
        "className": "user-dropdown",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "profile",
            "className": "user-dropdown__image",
            "src": "test.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "className": "user-dropdown__username",
          },
          "ref": null,
          "rendered": "test",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
